### PR TITLE
Fixed stave spacing

### PR DESCRIPTION
**Pull request by iv41879**
- Fixed stave spacing feature implemented
- New param distance added for \staffFormat tag

### CODE REVIEW

File: src/engine/abstract/ARStaffFormat.cpp [hunk 1]
```diff
@@ -49,7 +50,7 @@ void ARStaffFormat::setTagParameterList(TagParameterList & tpl)
 	if (ltpls.GetCount() == 0) {
 		// create a list of string ...
 		ListOfStrings lstrs; // (1); std::vector test impl
-		lstrs.AddTail(("S,style,standard,o;U,size,3pt,o;F,lineThickness,0.08,o"));
+        lstrs.AddTail(("S,style,standard,o;U,size,3pt,o;F,lineThickness,0.08,o;F,distance,0.0,o"));
```
Comment (dfober) [inline src/engine/abstract/ARStaffFormat.cpp]: you should rather use `U` as parameter type (i.e. unit)

File: src/engine/abstract/ARStaffFormat.cpp [hunk 2]
```diff
@@ -101,6 +102,13 @@ void ARStaffFormat::setTagParameterList(TagParameterList & tpl)
 				fLineThickness = (LSPACE * fval->getValue() >= 0 ? LSPACE * fval->getValue() : 0);
 
 			delete fval;
+
+            TagParameterFloat* distance = TagParameterFloat::cast(rtpl->RemoveHead());
+            assert(distance);
+            if (distance->TagIsSet())
+                fDistance = (LSPACE * distance->getValue() >= 0 ? LSPACE * distance->getValue() : 0);
```
Comment (dfober) [inline src/engine/abstract/ARStaffFormat.cpp]: don't assume the value is a LSPACE, conversion will be automatic if you're using a unit type

File: src/engine/graphic/GRSystemSlice.cpp [hunk 3]
```diff
@@ -372,6 +372,11 @@ void GRSystemSlice::Finish()
 			if (prev && !prev->getStaffState()->distanceset)
                 nextposition.y -= myrect.top;
 
+            // fixed stave spacing
+            float staffDistance = theStaff->getStaffState()->getStaffDistance();
+            if (staffDistance > 0.0)
+                nextposition.y = (theStaff->getDredgeSize() + staffDistance) * (i - 1);
```
Comment (dfober) [inline src/engine/graphic/GRSystemSlice.cpp]: nextposition must be relative to the previous staff
the following code fails with this implementation:
   ```
{  
    [ a b c ],
    [	\staffFormat<distance=2> a b c ],
    [	\staffFormat<distance=3> a b c ],
    [	\staffFormat<distance=2> a b c ]
   }
```

**Comment by dfober** [comment]
You should also add a new sample file in `gmn-examples/new-tags-v1.64` that illustrates the new parameter, including in combination with the other parameters. Test carefully with different settings and scores.

**Comment by iv41879** [comment]
Thanks @dfober  for review.

**Comment by dfober** [comment]
fine ! and it’s merged to dev
you should also add a comment to the ‘changeling.txt’ file
	https://github.com/grame-cncm/guidolib/blob/dev/changelog.txt <https://github.com/grame-cncm/guidolib/blob/dev/changelog.txt>
and please, add yourself to the contributors list:
	https://github.com/grame-cncm/guidolib/blob/dev/package/contributors.txt <https://github.com/grame-cncm/guidolib/blob/dev/package/contributors.txt>
Thanks
—
Dom

**Pull request merged**